src/lib/Libpython/pbs_python_svr_internal.c

```diff
@@ -2521,19 +2521,28 @@ pbs_python_populate_svrattrl_from_python_class(PyObject *py_instance,
 	PyObject	*py_resc_hookset_dict0 = (PyObject *)NULL;
 	PyObject	*py_attr_keys = (PyObject *)NULL;
 	PyObject 	*py_val = (PyObject *)NULL;
-	char	*name_str_dup = NULL;
-	char	*val_str_dup = NULL;
+	char		*name_str_dup = NULL;
+	char		*val_str_dup = NULL;
 	int		num_attrs, i;
 	pbs_list_head	svrattrl_list2;
-	int         rc = -1;
+	int		rc = -1;
 	int		hook_set_flag = 0;
 	int		has_resv_duration;
-	char            the_resc[HOOK_BUF_SIZE];
-	char            the_val[HOOK_BUF_SIZE];
+	char		the_resc[HOOK_BUF_SIZE];
+	static char     *the_val = NULL;
+	static int 	val_buf_size = HOOK_BUF_SIZE;
 	PyObject	*py_resc = (PyObject *)NULL;
 	long		val_sec;
 	char		*objname = NULL;
 
+	if (the_val == NULL) {
+		the_val = (char *)malloc(HOOK_BUF_SIZE*sizeof(char));
+		if (the_val == NULL) {
+			log_err(PBSE_SYSTEM, __func__, "Out of memory error");
+			return rc;
+		}
+	}
+
```

Comment:
pbs_strcat actually concatenates two string. So i passed an empty string before calling pbs_strcat instead of passing NULL, therefore did a malloc to initialize the it.